### DISCUSSION TITLE
We have a new MicroG

### DISCUSSION THREAD

**Discussion by ghost** [ReVanced/revanced-discussions#287] [General]
Deprecated

**Comment by oSumAtrIX** [comment]
ReVanced does not use MicroG.

**Reply by ghost** [reply]
And what use to login on non-root?

**Reply by oSumAtrIX** [reply]
YouTube is not ReVanced.

**Reply by oSumAtrIX** [reply]
Additionally, your repo does not contain any history from the original MicroG. One would have to go through the entire source files to be sure it is safe.

**Reply by ghost** [reply]
> Additionally, your repo does not contain any history from the original MicroG. One would have to go through the entire source files to be sure it is safe.

Well...I'll re-push it tomorrow with a fork from Vanced.

**Reply by ghost** [reply]
> Additionally, your repo does not contain any history from the original MicroG. One would have to go through the entire source files to be sure it is safe.

Done!

**Comment by nyxiereal** [comment]
When you would add support for other arcs? Then i night add support for uMicroG to my repo

**Reply by ghost** [reply]
Which arcs are you interested in?

**Reply by nyxiereal** [reply]
universal / nodpi

**Reply by ghost** [reply]
I created 2 branches on repo. Give it a look.

**Comment by ghost** [comment]
@xemulat Why don't make a release (like previous Vanced)? This will help to deprecate current (and bloated) Vanced MicroG.

**Reply by oSumAtrIX** [reply]
The ideal scenario would be to somehow get the necessary changes to MicroG by Vanced merged to MicroG upstream as this way people who use regular MicroG do not have to use Vanced MicroG. Forking the fork is making matters worse.

**Reply by bogadana** [reply]
upstream doesnt want those changes since they aim to replace stock google services

**Reply by oSumAtrIX** [reply]
Which means being on par with MicroG is better than completely branching off.

**Reply by ghost** [reply]
> The ideal scenario would be to somehow get the necessary changes to MicroG by Vanced merged to MicroG upstream as this way people who use regular MicroG do not have to use Vanced MicroG. Forking the fork is making matters worse.

MicroG creator hate Vanced and every related project.

I doubt he'll accept this.

Was be always a problem for Vanced team.

**Reply by oSumAtrIX** [reply]
Why should they?

**Reply by ghost** [reply]
> Why should they?

For a really easy reason: the creator prefer to collaborate just with entirely open source projects, like NewPipe:

Check this discussion for further info: https://github.com/microg/GmsCore/issues/1279

**Reply by oSumAtrIX** [reply]
Vanced MicroG is entirely open source, what do you mean.

**Reply by ghost** [reply]
> Vanced MicroG is entirely open source, what do you mean.

That MicroG creator believes that apps like Vanced (and therefore ReVanced) are useless.

Read the discussion.

**Reply by oSumAtrIX** [reply]
ReVanced is completely open-source, there's no room for false beliefs, additionally, we are currently talking about MicroG, not Vanced, and not ReVanced.

**Reply by ghost** [reply]
> ReVanced is completely open-source, there's no room for false beliefs, additionally, we are currently talking about MicroG, not Vanced, and not ReVanced.

You read the discussion. Marv-in will not change his mind. If you are sure otherwise, open an issue and talk with him.

**Reply by Lyceris-chan** [reply]
the help channel in the discord clearly states the following: 
XIII 🔸 Will MicroG stay up to date?
MicroG and Vanced MicroG are under development by their respective maintainer.

Besides that for all I can tell MicroG hasn't received any major changes which would be required to preserve and or improve upon its current functionality when compared to the latest VancedMicroG build so I really don't see a reason as to why this discussion even started.

**Reply by oSumAtrIX** [reply]
The question is if we want to diverge away from upstream completely or not. I would say its better to be as near to upstream as possible to be able to merge as fluently as possible.

**Reply by ghost** [reply]
> Vanced MicroG are under development by their respective maintainer.

Are you sure of that?

**Reply by Lyceris-chan** [reply]
idk if its true, blame the dude who wrote the stuff in the help channel if it isn't lmao

**Reply by anil-al** [reply]
In any way, there is a clear fact, microG should be updated since it keeps stopping in the background about 3 times a day in the last release which wasn't the case before a couple of weeks ago. I use it in two different device and both of them have the same problem. One way or another, it seems there is a need for a new release.

**Reply by oSumAtrIX** [reply]
Disable battery optimization and it should not happen.

**Reply by anil-al** [reply]
I've done it already. That's the first thing the app said you to do so. Something is wrong about the app.

**Reply by pascua28** [reply]
> The ideal scenario would be to somehow get the necessary changes to MicroG by Vanced merged to MicroG upstream as this way people who use regular MicroG do not have to use Vanced MicroG. Forking the fork is making matters worse.

Unless you are talking about the root variant of ReVanced YouTube, this is just not possible. The reason is that the original MicroG uses `com.google.android.gms` as its package name, which means, it's meant as a Google Play Services replacement for custom ROMs that do not have Google Play Services. Therefore, it's not possible to install it on non-root users. That's why Vanced team renamed the package name, along with some of the methods from `com.google.*` to `com.mgoogle.*`. That way, non-root users will be able to install the modified MicroG and allow it to co-exist with Google Play Services
As for the Vanced YouTube app, it's not possible to install it using the same package name for non-root users, that's why we have to rename it to something like `revanced.app.android.youtube`. The resulting problem, however, of renaming the YouTube package name is that we are not allowed to use it alongside Google Play Services so it has to use a modified MicroG.
By using root variant, on the other hand, we can just replace the original YouTube app with the modified app while retaining the same package name, and so, Vanced MicroG is not needed anymore as it can use Google Play Services. But not everyone would want to root their phones

**Reply by oSumAtrIX** [reply]
> The reason is that the original MicroG uses com.google.android.gms as its package name

I was referring to all mods which are not necessary such as removing code, making small changes to support devices by different vendors or fixing broken code by unsupported means. These are changes which branch off the development of MicroG and thus are out of the question to get merged. Necessary changes such as the renaming of the packages are certain patches you'd do on top of upstream along upstream, but small changes, if done properly, are ideally subject for a PR to upstream.

**Reply by pascua28** [reply]
> > The reason is that the original MicroG uses com.google.android.gms as its package name
> 
> I was referring to all mods which are not necessary such as removing code, making small changes to support devices by different vendors or fixing broken code by unsupported means. These are changes which branch off the development of MicroG and thus are out of the question to get merged.

If you have MicroG installed and working, that means you already have root. Vanced root variant will just work even without Vanced MicroG. It just needs Google Services, whether from vanilla MicroG or Google Play Services
For a non-root method, there's no other way to get vanilla MicroG and Vanced YouTube work other than to make a fork of MicroG

> Necessary changes such as the renaming of the packages are certain patches you'd do on top of upstream along upstream, but small changes, if done properly, are ideally subject for a PR to upstream.

For a non-rooted device, renaming of packages is necessary. Let's say you have managed to rename the package and some methods to `com.mgoogle.*`, installed and got MicroG working, all other apps that use Google Services will not recognize MicroG anymore and won't have Google Services functionality which defeats the purpose of MicroG.

**Reply by oSumAtrIX** [reply]
Yes, either non-root + Vanced MicroG or root & (Vanced) MicroG. Still, bug fixes concern MicroG and are not just related to Vanced. So if you have to make changes to MicroG for Vanced which are applicable to MicroG you should PR it so it is available in upstream and you do not have to maintain it anymore. Only changes that concern Vanced now have to stay in sync with upstream.

**Reply by pascua28** [reply]
> Yes, either non-root + Vanced MicroG or root & (Vanced) MicroG. Still, bug fixes concern MicroG and are not just related to Vanced. So if you have to make changes to MicroG for Vanced which are applicable to MicroG you should PR it so it is available in upstream and you do not have to maintain it anymore. Only changes that concern Vanced now have to stay in sync with upstream.

Yeah that would be good. It's why I also forked microG and added some minimal changes needed for Vanced so in the future, it would be easier for me to merge microG changes: https://github.com/pascua28/VancedGmsCore

**Comment by pascua28** [comment]
Vanced microG causes unusual battery drain on my device so I made my own personal build for Vanced. This is a work in progress and there are still services/components that I will remove in the future. Feel free to check it out here: https://github.com/pascua28/VancedMicroG

**Reply by TheJeterLP** [reply]
> Vanced microG causes unusual battery drain on my device so I made my own personal build for Vanced. This is a work in progress and there are still services/components that I will remove in the future. Feel free to check it out here: https://github.com/pascua28/VancedMicroG

It does not cause unusual battery drain. Only certain features like casting do. That's why they're blocked to use.

**Reply by oSumAtrIX** [reply]
Please fork the original project instead of creating a new repo for it, we cant see what you changed, something in the history could be altered.

**Reply by pascua28** [reply]
> Please fork the original project instead of creating a new repo for it, we cant see what you changed, something in the history could be altered.

Ok. Here: https://github.com/pascua28/VancedGmsCore

**Reply by pascua28** [reply]
I just decided to compile my own also because the git history of VancedMicroG is somewhat messy and I have some trust issues

**Reply by TheJeterLP** [reply]
> I just decided to compile my own also because the git history of VancedMicroG is somewhat messy and I have some trust issues

Trust issues? What makes you get these?

**Reply by pascua28** [reply]
> > I just decided to compile my own also because the git history of VancedMicroG is somewhat messy and I have some trust issues
> 
> Trust issues? What makes you get these?

I don't know, lol

**Reply by TheJeterLP** [reply]
> > > I just decided to compile my own also because the git history of VancedMicroG is somewhat messy and I have some trust issues
> > 
> > Trust issues? What makes you get these?
> 
> I don't know, lol

Err, okay I guess? Vanced MicroG is a fork of MicroG. So you can see all the changes they made. No trust issues here at all with them

**Reply by pascua28** [reply]
> > > > I just decided to compile my own also because the git history of VancedMicroG is somewhat messy and I have some trust issues
> > > 
> > > 
> > > Trust issues? What makes you get these?
> > 
> > 
> > I don't know, lol
> 
> Err, okay I guess? Vanced MicroG is a fork of MicroG. So you can see all the changes they made. No trust issues here at all with them

As this is a personal project, I would prefer to keep a cleaner history so I could pull updates and build on my own
